### ISSUE TITLE
using setupFiles from parent dir causes 'Failed to load url'

### ISSUE THREAD

**Issue by unrealsolver** [vitest-dev/vitest#5723]
### Describe the bug

Having this option in vite.config.ts `setupFiles: '../vitest.setup',` together with `package.json` (considering monorepo) produces an error:
```
 FAIL  src/lib/integrations.spec.tsx [ src/lib/integrations.spec.tsx ]
Error: Failed to load url /home/x/y/nt2/vitest.setup.ts (resolved id: /home/x/y/nt2/vitest.setup.ts). Does the file exist?
```
Removing package.json resolves the issue.
Context: https://github.com/nrwl/nx/issues/21259

### Reproduction

```
npx create-nx-workspace vtdemo --preset=react-monorepo --appName=app --bundler=vite --e2eTestRunner=none --style=css --nxCloud=skip
cd vtdemo
npx nx g @nx/react:lib other --buildable --unitTestRunner=vitest --bundler=vite
touch vitest.setup.ts
nvim other/vite.config.ts
```
Now put `setupFiles: '../vitest.setup',` into `test` entry
Now run
```
npx nx test other
```
Observe the error.
This does not depend on yarn/npm. This only depend on the `package.json` existence inside the `other` package. If there are no `package.json` (like there are no for nx apps) it works fine. Nx dev team says this is vitest regression. 
Context: https://github.com/nrwl/nx/issues/21259

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
    Memory: 8.72 GB / 31.09 GB
    Container: Yes
    Shell: 3.3.1 - /usr/bin/fish
  Binaries:
    Node: 18.20.2 - ~/.local/share/nvm/v18.20.2/bin/node
    Yarn: 4.2.2 - ~/.local/share/nvm/v18.20.2/bin/yarn
    npm: 10.5.0 - ~/.local/share/nvm/v18.20.2/bin/npm
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
This behavior follows Vite's `fs.strict` policy if you are using browser-like environments: https://vitejs.dev/config/server-options.html#server-fs-strict

You need to have the root as the root of the project or add the setup file to `fs.allow`.

**Comment by unrealsolver** [comment]
Indeed, this is Vite feature that has very confusing error message. I'll put updates to Nx issue discussion.
